turbo/cli/make_app.go

```diff
@@ -24,5 +33,99 @@ func MakeApp(action func(*cli.Context), cliFlags []cli.Flag) *cli.App {
 		debug.Exit()
 		return nil
 	}
+	app.Commands = []cli.Command{initCommand}
 	return app
 }
+
+var initCommand = cli.Command{
+	Action:    MigrateFlags(initGenesis),
+	Name:      "init",
+	Usage:     "Bootstrap and initialize a new genesis block",
+	ArgsUsage: "<genesisPath>",
+	Flags: []cli.Flag{
+		utils.DataDirFlag,
+	},
+	Category: "BLOCKCHAIN COMMANDS",
+	Description: `
+The init command initializes a new genesis block and definition for the network.
+This is a destructive action and changes the network in which you will be
+participating.
+
+It expects the genesis file as argument.`,
+}
+
+// initGenesis will initialise the given JSON format genesis file and writes it as
+// the zero'd block (i.e. genesis) or will fail hard if it can't succeed.
+func initGenesis(ctx *cli.Context) error {
+	// Make sure we have a valid genesis JSON
+	genesisPath := ctx.Args().First()
+	if len(genesisPath) == 0 {
+		utils.Fatalf("Must supply path to genesis JSON file")
+	}
+
+	file, err := os.Open(genesisPath)
+	if err != nil {
+		utils.Fatalf("Failed to read genesis file: %v", err)
+	}
+	defer file.Close()
+
+	genesis := new(core.Genesis)
+	if err := json.NewDecoder(file).Decode(genesis); err != nil {
+		utils.Fatalf("invalid genesis file: %v", err)
+	}
+
+	// Open and initialise both full and light databases
+	stack := MakeConfigNodeDefault()
+	defer stack.Close()
+
+	for _, name := range []string{"chaindata", "lightchaindata"} {
+		chaindb, err := node.OpenDatabase(stack.Config(), log.New(ctx), kv.ChainDB)
+		if err != nil {
+			utils.Fatalf("Failed to open database: %v", err)
+		}
+		_, hash, err := core.CommitGenesisBlock(chaindb, genesis)
+		if err != nil {
+			utils.Fatalf("Failed to write genesis block: %v", err)
+		}
+		chaindb.Close()
+		log.Info("Successfully wrote genesis state", "database", name, "hash", hash.Hash())
+	}
+	return nil
+}
+
+func MigrateFlags(action func(ctx *cli.Context) error) func(*cli.Context) error {
+	return func(ctx *cli.Context) error {
+		for _, name := range ctx.FlagNames() {
+			if ctx.IsSet(name) {
+				ctx.GlobalSet(name, ctx.String(name))
+			}
+		}
+		return action(ctx)
+	}
+}
+
+func NewNodeConfig() *node.Config {
+	nodeConfig := node.DefaultConfig
+	// see simiar changes in `cmd/geth/config.go#defaultNodeConfig`
+	if commit := params.GitCommit; commit != "" {
+		nodeConfig.Version = params.VersionWithCommit(commit, "")
+	} else {
+		nodeConfig.Version = params.Version
+	}
+	nodeConfig.IPCPath = "erigon.ipc" // force-disable IPC endpoint
```

Comment:
you need somehow use utils.DataDirFlag which you defined above
normally it's happening in `cmd/utils/flag.go:SetNodeConfig->setDataDir`